applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py

```diff
@@ -32,9 +32,16 @@ def __init__(self, Model, settings):
         self.epsilon = settings["epsilon"].GetDouble()
 
         self.fluid_model_part = self.body_model_part.GetRootModelPart()
-        self.trailing_edge_model_part = self.fluid_model_part.CreateSubModelPart("trailing_edge_model_part")
-        #List to store trailing edge elements id
+        if not self.fluid_model_part.HasSubModelPart("trailing_edge_model_part"):
+            self.trailing_edge_model_part = self.fluid_model_part.CreateSubModelPart("trailing_edge_model_part")
+        else: self.trailing_edge_model_part = self.fluid_model_part.GetSubModelPart("trailing_edge_model_part")
+
+        if not self.fluid_model_part.HasSubModelPart("wake__elements"):
+            self.wake_modelpart = self.fluid_model_part.CreateSubModelPart("wake__elements")
```

Comment:
I would rename this to:

```suggestion
            self.wake_sub_model_part = self.fluid_model_part.CreateSubModelPart("wake__elements")
```

Reply:
in the fsi coupling loop we need to calculate the forces applied, on nodes, in order to transfer them to the structure. If we use the FinializeSolutionStep function then it won't do the job during the coupling and we would get Zero forces on the structure
Using FinalizeNonLinearIteration, we could use the values now in the SolveSolutionStep function inside kratos_potential_flow_solver